CMakeLists.txt

```diff
@@ -257,7 +257,6 @@ endfunction(assign_source_group)
 
 # Adjust settings for shared libraries if requested
 if(MATERIALX_BUILD_SHARED_LIBS)
-    add_definitions(-DMATERIALX_BUILD_SHARED_LIBS)
     set(BUILD_SHARED_LIBS "ON")
```

Comment:
In [Library.h](https://github.com/autodesk-forks/MaterialX/blob/adsk_contrib/dev/source/MaterialXCore/Library.h#L23) we check for `#if defined(MATERIALX_BUILD_SHARED_LIBS)` so the compiler definition needs be there or not since we are checking for presence instead of value.

We do not need to set it globally, because we make them CMake `PUBLIC` in each library that has exported functions. This means the define is added when compiling the library, and is automatically set for each downstream client using the library (including internal downstream clients like MaterialXView).

Reply:
Yes, we do need some define in the code but we also introduce another cmake value to test instead of just BUILD_SHARED_LIBS. Wondering why we don't just use that. It's more a nitpick to not introduce custom cmake vars.

As for the second, if it's set for downstream clients, can we just set it just in MaterialXCore ?

Reply:
The BUILD_SHARED_LIBS cmake var is for NanoGUI. I will update the comment.